service/matching/db_task_manager.go

```diff
@@ -164,14 +169,25 @@ func (d *dbTaskManager) Start() {
 	}
 
 	d.signalDispatch()
-
-	// TODO: consider using goro.Group
-	go d.acquireLoop(context.Background())
-	go d.writerEventLoop(context.Background())
-	go d.readerEventLoop(context.Background())
+	d.group.Go(d.acquireLoop)
+	d.group.Go(d.writerEventLoop)
+	d.group.Go(d.readerEventLoop)
 }
 
 func (d *dbTaskManager) Stop() {
+	defer d.group.Wait()
```

Comment:
why we need this?

Reply:
Don't we want to make sure all its go routines exit before db task manager is deallocated?